### PR TITLE
Travis: Update to xenial

### PR DESCRIPTION
**Pull request by jdinan**
Yay!!  --- https://blog.travis-ci.com/2018-11-08-xenial-release

### CODE REVIEW

**Comment by jdinan** [comment]
Well, this is working now.  We have 1:30 margin on the CMA row in the testing matrix.  But, we also had to make some concessions in terms of testing coverage to get there.  Should we merge this?

**Comment by davidozog** [comment]
I think it's ok to merge, especially if these tests are covered by Jenkins, which I believe they are.  Is that the case @wrrobin?

**Comment by jdinan** [comment]
@davidozog @wrrobin This should be ready for review.  I refactored large parts of the testing script to handle OFI, Portals 4, and PMIx through separate rows in the testing matrix.  There should be no meaningful reduction in coverage now and hopefully should now be much easier to add to the build matrix.  Also learned a painful lesson about libev/libevent along the way.

**Comment by jdinan** [comment]
Also, these changes leave enough headroom that we can let the full CMA testing row run without any hacks to reduce testing time.

**Comment by jdinan** [comment]
One last update to add XPMEM testing to the Travis script.

**Comment by jdinan** [comment]
Cleared testing, xpmem and all!  :metal:

**Comment by jdinan** [comment]
@davidozog @wrrobin This is ready for review (promise to stop tinkering now).  It's a long read, so I'm happy to walk you through it at tomorrow's team meeting.

Portals, CMA, and PMIx now have separate rows in the build matrix (they used to be mixed in with every build).  This sped up the individual rows substantially.  Splitting things out and getting dependencies from the package manager shortened the logs a lot, so we should not need to view the raw logs.  Portals UDP build is still too buggy to run the tests; it quits the testing cycle early.  I also added XPMEM to the build matrix (yay!).

File: .travis.yml [hunk 1]
```diff
@@ -107,41 +172,52 @@ before_install:
...
-    - wget http://www.mpich.org/static/downloads/3.2/hydra-3.2.tar.gz
-    - tar xvzf hydra-3.2.tar.gz
-    - cd hydra-3.2/
-    - ./configure --prefix=$TRAVIS_INSTALL/hydra
-    - make $TRAVIS_PAR_MAKE
-    - make install
+    - >
+      if [[ $SOS_TRANSPORT_OPTS = *"with-ofi"* ]]; then
+        cd $TRAVIS_SRC
+        git clone -b v1.6.x --depth 10 https://github.com/ofiwg/libfabric.git libfabric
```
Comment (davidozog) [inline .travis.yml]: Should we move to v1.7.x?  I seem to recall we've discussed doing this.

File: .travis.yml [hunk 2]
```diff
@@ -107,41 +172,52 @@ before_install:
...
+        ## Install XPMEM
+        cd $TRAVIS_SRC
+        git clone --depth 10 https://github.com/hjelmn/xpmem.git
+        cd xpmem
+        ./autogen.sh
+        ./configure --prefix=$TRAVIS_INSTALL/xpmem
+        make $TRAVIS_PAR_MAKE
+        make install
+        ## Hack to get SOS to see the XPMEM library
+        ln -s $TRAVIS_INSTALL/xpmem/lib $TRAVIS_INSTALL/xpmem/lib64
```
Comment (davidozog) [inline .travis.yml]: I wonder what the cause of this is... I swear I've had to do this before, but I cannot remember which library it was.  Is it a libtool bug?

File: .travis.yml (hunk 1 above)
Comment (jdinan) [reply .travis.yml]: Sure, mind if I do that as a follow-up PR?

File: .travis.yml (hunk 2 above)
Comment (jdinan) [reply .travis.yml]: SOS configury looks for the library in the $WITH_XPMEM/lib64.  🤦‍♂️

**Comment by wrrobin** [comment]
@jdinan All the changes look good to me. Also, a discussion tomorrow on the changes will be helpful as well. Looking at the last round of tests, the first few of them did run 3x faster compared to others. Is there any particular reason?

File: .travis.yml (hunk 2 above)
Comment (jdinan) [reply .travis.yml]: Ref #685

**Comment by jdinan** [comment]
@wrrobin Those are the Portals tests.  They build and check SOS, then quit before running any of the external tests (UH, Cray, Mellanox, Parres, etc.).

File: .travis.yml (hunk 1 above)
Comment (davidozog) [reply .travis.yml]: Of course.

File: .travis.yml (hunk 2 above)
Comment (davidozog) [reply .travis.yml]: Haha!  I thought I'd seen this before.  Maybe that library was actually xpmem...  😖

**Comment by wrrobin** [comment]
Ok. I missed that they did only build and not ran any test.

**Comment by davidozog** [comment]
Now that I think of it, I'm surprised Portals over sockets used to pass all those test suites but not SOS tests...  Maybe we want to keep one (or more) of them?

**Comment by wrrobin** [comment]
The current setup is fine with me though. We can add more Portals tests once we need to check them for any reason. Adding any one also is ok.

**Comment by jdinan** [comment]
The old Travis script built the external tests with Portals, but did not run them.

**Comment by davidozog** [comment]
Ah yes.  I see now that none of the suites were actually run with Portals.  I was misled by the old comment, "`Run the UH test suite (Portals)`"...

**Comment by jdinan** [comment]
Yeah, those comments were very aspirational.  If we ever get librxe (soft RoCE) working, we might be able to run the verbs build of Portals in Travis.  (I checked, it's not a simple package install in Xenial)

**Comment by jdinan** [comment]
Sounds like @davidozog is good with merging.  @wrrobin Go ahead or wait for tomorrow?

**Comment by wrrobin** [comment]
@jdinan Sorry, i thought i approved it earlier.

**Pull request merged**